cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/store/remote/RemoteNamespaceQueryTest.java

```diff
@@ -16,47 +16,59 @@
 
 package co.cask.cdap.internal.app.store.remote;
 
+import co.cask.cdap.common.conf.CConfiguration;
 import co.cask.cdap.common.namespace.RemoteNamespaceQueryClient;
+import co.cask.cdap.data2.datafabric.dataset.service.DatasetService;
 import co.cask.cdap.gateway.handlers.meta.RemoteNamespaceQueryHandler;
-import co.cask.cdap.internal.app.services.http.AppFabricTestBase;
-import co.cask.cdap.proto.Id;
+import co.cask.cdap.gateway.handlers.meta.RemoteSystemOperationsService;
+import co.cask.cdap.internal.guice.AppFabricTestModule;
 import co.cask.cdap.proto.NamespaceConfig;
 import co.cask.cdap.proto.NamespaceMeta;
 import co.cask.cdap.proto.id.NamespaceId;
 import co.cask.cdap.store.NamespaceStore;
+import co.cask.tephra.TransactionManager;
+import com.google.inject.Guice;
 import com.google.inject.Injector;
+import org.junit.AfterClass;
 import org.junit.Assert;
 import org.junit.BeforeClass;
+import org.junit.ClassRule;
 import org.junit.Test;
-
-import java.util.List;
+import org.junit.rules.TemporaryFolder;
 
 /**
  * Tests implementation of {@link RemoteNamespaceQueryHandler} by using it fetch namespaces.
  */
-public class RemoteNamespaceQueryTest extends AppFabricTestBase {
+public class RemoteNamespaceQueryTest {
+  @ClassRule
+  public static final TemporaryFolder TEMPORARY_FOLDER = new TemporaryFolder();
+
+  private static TransactionManager txManager;
+  private static DatasetService datasetService;
+  private static RemoteSystemOperationsService remoteSysOpService;
 
   private static NamespaceStore namespaceStore;
   private static RemoteNamespaceQueryClient queryClient;
 
   @BeforeClass
   public static void setup() throws Exception {
-    Injector injector = getInjector();
+    CConfiguration cConf = CConfiguration.create();
+    Injector injector = Guice.createInjector(new AppFabricTestModule(cConf));
+    txManager = injector.getInstance(TransactionManager.class);
+    txManager.startAndWait();
+    datasetService = injector.getInstance(DatasetService.class);
+    datasetService.startAndWait();
+    remoteSysOpService = injector.getInstance(RemoteSystemOperationsService.class);
+    remoteSysOpService.startAndWait();
```

Comment:
sorry, one more thing. I saw a recent failure in RemotePrivilegesFetcher test where a request was made before the service was discoverable, which leads to flakiness. https://github.com/caskdata/cdap/pull/6090/commits/e7a8505a2575a15827db5541f1b115511a8a6d07. You may want to guard against it too.